docroot/modules/custom/sitenow_signage/sitenow_signage.module

```diff
@@ -338,34 +323,124 @@ function sitenow_signage_preprocess_page(&$variables) {
 function sitenow_signage_preprocess_paragraph(&$variables) {
   $paragraph = $variables['paragraph'];
 
-  if ($paragraph->bundle() == 'slide_image' &&
-    isset($variables['content']['field_slide_image'])) {
+  switch ($paragraph->bundle()) {
+    case 'slide_image':
+      if (isset($variables['content']['field_slide_image'])) {
 
-    // Get the sign node from the current route.
-    $sign_node = \Drupal::routeMatch()->getParameter('node');
+        // Get the sign node from the current route.
+        $sign_node = \Drupal::routeMatch()->getParameter('node');
 
-    if ($sign_node && $sign_node->bundle() == 'sign') {
-      $variables['#cache']['tags'][] = 'node:' . $sign_node->id();
-    }
+        if ($sign_node && $sign_node->bundle() == 'sign') {
+          $variables['#cache']['tags'][] = 'node:' . $sign_node->id();
+        }
+
+        // Check if we have a sign node with vertical orientation.
+        if ($sign_node &&
+          $sign_node->bundle() == 'sign' &&
+          $sign_node->hasField('field_sign_orientation') &&
+          !$sign_node->get('field_sign_orientation')->isEmpty() &&
+          $sign_node->get('field_sign_orientation')->value === 'vertical') {
+
+          $view_builder = \Drupal::entityTypeManager()->getViewBuilder('media');
+
+          foreach ($variables['content']['field_slide_image'] as $delta => &$item) {
+            if (is_numeric($delta)) {
+              $media_entity = $paragraph->get('field_slide_image')->get($delta)->entity;
+              if ($media_entity) {
+                $item = $view_builder->view($media_entity, 'vertical_9_16');
+              }
+            }
+          }
+        }
+      }
+      break;
 
-    // Check if we have a sign node with vertical orientation.
-    if ($sign_node &&
-      $sign_node->bundle() == 'sign' &&
-      $sign_node->hasField('field_sign_orientation') &&
-      !$sign_node->get('field_sign_orientation')->isEmpty() &&
-      $sign_node->get('field_sign_orientation')->value === 'vertical') {
+    case 'slide_events':
+      $paragraph = $variables['elements']['#paragraph'];
+      $filters = [];
 
-      $view_builder = \Drupal::entityTypeManager()->getViewBuilder('media');
+      $headline = $paragraph?->field_headline?->value ?? '';
 
-      foreach ($variables['content']['field_slide_image'] as $delta => &$item) {
-        if (is_numeric($delta)) {
-          $media_entity = $paragraph->get('field_slide_image')->get($delta)->entity;
-          if ($media_entity) {
-            $item = $view_builder->view($media_entity, 'vertical_9_16');
-          }
+      $filters['audiences'] = sitenow_signage_value_query_string($paragraph?->field_slide_events_audiences?->getValue());
```

Comment:
```suggestion
      $filters['audiences'] = sitenow_signage_value_query_string($paragraph?->field_slide_events_audiences?->value);
```
Again, this is typically how we do this type of accessor unless there is a specific reason to use the `getValue()` method.

Reply:
The `->value` accessor only gets the first value of a multi option field. `->getValue()` here allows us to get an array of all the selected options.